src/stdlib_math.fypp

```diff
@@ -1,21 +1,270 @@
 #:include "common.fypp"
 #:set IR_KINDS_TYPES = INT_KINDS_TYPES + REAL_KINDS_TYPES
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
 
 module stdlib_math
     use stdlib_kinds, only: int8, int16, int32, int64, sp, dp, qp
 
     implicit none
     private
-    public :: clip
+    public :: clip, linspace, logspace
+    public :: EULERS_NUMBER_SP, EULERS_NUMBER_DP, EULERS_NUMBER_QP
+    public :: DEFAULT_LINSPACE_LENGTH, DEFAULT_LOGSPACE_BASE, DEFAULT_LOGSPACE_LENGTH
+
+    integer, parameter :: DEFAULT_LINSPACE_LENGTH = 100
+    integer, parameter :: DEFAULT_LOGSPACE_LENGTH = 50
+    integer, parameter :: DEFAULT_LOGSPACE_BASE = 10
+
+    ! Useful constants for lnspace
+    real(sp), parameter :: EULERS_NUMBER_SP = 2.71828182845904523536028747135266249775724709369995_sp
+    real(dp), parameter :: EULERS_NUMBER_DP = 2.71828182845904523536028747135266249775724709369995_dp
+    real(qp), parameter :: EULERS_NUMBER_QP = 2.71828182845904523536028747135266249775724709369995_qp
 
     interface clip
     #:for k1, t1 in IR_KINDS_TYPES
         module procedure clip_${k1}$
     #:endfor
     end interface clip
 
+    interface linspace
+    !! Version: Experimental
+    !!
+    !! Create rank 1 array of linearly spaced elements
+    !! If the number of elements is not specified, create an array with size 100. If n is a negative value,
+    !! return an array with size 0. If n = 1, return an array whose only element is end
+    !!([Specification](../page/specs/stdlib_math.html#description))
```

Comment:
```suggestion
    !!([Specification](../page/specs/stdlib_math.html#linspace-create-a-linearly-spaced-rank-one-array))
```

Reply:
I apologize for this. I didn't totally understand how the html tags got incorporated after uploading the specs, and I sorta just copied how the specs appeared in a different section of stdlib.

Reply:
No problem with that. Before approving a PR, I usually load the results of the github actions, and check if every new link is working fine.